src/threading.h

```diff
@@ -632,19 +633,13 @@ template<bool is_write, class type> void prefetch(const type& p) {
 template<class type> void prefetch_write(const type& p) { prefetch<true>(p); }
 template<class type> void prefetch_read(const type& p) { prefetch<false>(p); }
 
-void memory_barrier() {
-    asm volatile( "" ::: "memory" );
-}
```

Comment:
for reference. This is not sufficient to have atomic access or cache synchronization. Back before compilers were good at optimizing and you could assume certain code generation on x86 with its guarantees, this is what you'd do. This *only* tells the compiler not to move loads and stores out of order (across the barrier). The CPU and the cache effects can still have the effect of re-ordering. That's why there needs to be language facilities that all levels of the stack honors. The compiler and the CPU.